pkg/container-utils/containerd/containerd.go

```diff
@@ -15,31 +15,246 @@
 	if socketPath == "" {
 		socketPath = runtimeclient.ContainerdDefaultSocketPath
 	}
 
-	criClient, err := criclient.NewCRIClient(runtimeclient.ContainerdName, socketPath, DefaultTimeout)
+	client, err := containerd.New(socketPath,
+		containerd.WithTimeout(DefaultTimeout),
+		containerd.WithDefaultNamespace("k8s.io"),
+	)
 	if err != nil {
 		return nil, err
 	}
 
 	return &ContainerdClient{
-		CRIClient: criClient,
+		client: client,
+	}, nil
+}
+
+func (c *ContainerdClient) Close() error {
+	if c.client != nil {
+		return c.client.Close()
+	}
+	return nil
+}
+
+func (c *ContainerdClient) GetContainers() ([]*runtimeclient.ContainerData, error) {
+	containers, err := c.client.Containers(context.TODO(), "")
+	if err != nil {
+		return nil, fmt.Errorf("listing containers: %w", err)
+	}
+
+	ret := make([]*runtimeclient.ContainerData, 0, len(containers))
+	for _, container := range containers {
+		if isSandboxContainer(container) {
+			log.Debugf("ContainerdClient: container %q is a sandbox container. Temporary skipping it", container.ID())
+			continue
+		}
+
+		task, err := getContainerdTask(container)
+		if err != nil {
+			return nil, err
+		}
+
+		containerData, err := taskAndContainerToContainerData(task, container)
+		if err != nil {
+			return nil, err
+		}
+
+		ret = append(ret, containerData)
+	}
+
+	return ret, nil
+}
+
+func (c *ContainerdClient) GetContainer(containerID string) (*runtimeclient.ContainerData, error) {
+	containerID, err := runtimeclient.ParseContainerID(runtimeclient.ContainerdName, containerID)
+	if err != nil {
+		return nil, err
+	}
+
+	containerData, _, _, err := c.getContainerDataAndContainerAndTask(containerID)
+	if err != nil {
+		return nil, err
+	}
+	return containerData, nil
+}
+
+func (c *ContainerdClient) GetContainerDetails(containerID string) (*runtimeclient.ContainerDetailsData, error) {
+	containerID, err := runtimeclient.ParseContainerID(runtimeclient.ContainerdName, containerID)
+	if err != nil {
+		return nil, err
+	}
+
+	containerData, container, task, err := c.getContainerDataAndContainerAndTask(containerID)
+	if err != nil {
+		return nil, err
+	}
+
+	spec, err := container.Spec(context.TODO())
+	if err != nil {
+		return nil, fmt.Errorf("getting spec for container %q: %w", containerID, err)
+	}
+
+	mountData := make([]runtimeclient.ContainerMountData, len(spec.Mounts))
+	for i := range spec.Mounts {
+		mount := spec.Mounts[i]
+		mountData[i] = runtimeclient.ContainerMountData{
+			Source:      mount.Source,
+			Destination: mount.Destination,
+		}
+	}
+
+	return &runtimeclient.ContainerDetailsData{
+		ContainerData: *containerData,
+		Pid:           int(task.Pid()),
+		CgroupsPath:   spec.Linux.CgroupsPath,
+		Mounts:        mountData,
 	}, nil
 }
+
+func (c *ContainerdClient) getContainerDataAndContainerAndTask(containerID string) (*runtimeclient.ContainerData, containerd.Container, containerd.Task, error) {
```

Comment:
nit: I'm not a big fan of this function, so I would remove it as it is called twice and one of them is discarding most of the returned values. Anyway, it is more code styling, so it is up to you.

Reply:
What i wanted to avoid to these 3 calls to have in both locations. Doesn't matter for me too (Since it is only used twice)

I also don't like that one of the two usages discard everything except the `containerData` :sweat_smile: